### PR TITLE
FAI-8029 - Sheet Survey converter and Surveys Refactor

### PR DESCRIPTION
**Pull request by matiaslcoulougian**
## Description
- Add sheetId and recordId to Sheet source records
- Add Sheets surveys converter
- Move Surveys converter common logic from airtable/surveys specific converter to common surveys class. Inject class to Airtable and Sheets surveys converters to use the common functionalities.

## Type of change
- [ ] Bug fix
- [x] New feature
- [ ] Breaking change

## Related issues

> Fix [#1]() 

## Migration notes

> Describe migration notes if any

## Extra info

> Add any additional information

### CODE REVIEW

**Comment by matiaslcoulougian** [comment]
Should adapt changes to use https://github.com/faros-ai/airbyte-connectors/pull/1192 solution

File: destinations/airbyte-faros-destination/src/converters/airtable/surveys.ts [hunk 1]
```diff
@@ -41,6 +43,15 @@ export class Surveys extends AbstractSurveys {
   }
 
   getSubmittedAt(record: AirbyteRecord): string | undefined {
```
Comment (ypc-faros) [inline destinations/airbyte-faros-destination/src/converters/airtable/surveys.ts:45]: I think here we should return the value if present regardless of what the column name is. And fallback to `_airtable_created_time`

We'll have to document this behavior

File: destinations/airbyte-faros-destination/src/converters/sheets/surveys.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+import {AbstractSurveys} from '../abstract-surveys/surveys';
+
+export class Surveys extends AbstractSurveys {
+  source = 'Sheets';
+
+  id(record: AirbyteRecord): any {
+    return record?.record?.data?.id;
+  }
+
+  getSubmittedAt(record: AirbyteRecord): string | undefined {
```
Comment (ypc-faros) [inline destinations/airbyte-faros-destination/src/converters/sheets/surveys.ts:12]: We can actually make this the default implementation in `AbstractSurveys` and then in the Airtable one we can do `super.getSubmittedAt(x) ?? record?.record?.data?._airtable_created_time`

File: destinations/airbyte-faros-destination/src/converters/abstract-surveys/surveys.ts [hunk 3]
```diff
@@ -47,10 +48,17 @@ export interface SurveysConfig {
   column_names_mapping?: ColumnNameMapping;
 }
 
+export const RESPONSE_SUBMITTED_AT_DEFAULT_COLUMN_NAME = 'Timestamp';
```
Comment (ypc-faros) [inline destinations/airbyte-faros-destination/src/converters/abstract-surveys/surveys.ts:51]: Remove if not needed

File: sources/sheets-source/resources/schemas/sheets.json [hunk 4]
```diff
@@ -2,9 +2,16 @@
...
   "properties": {
+    "sheetId": {
+      "type": "string"
+    },
     "sheetName": {
       "type": "string"
     },
+    "id": {
+      "type": "string",
+      "description": "The id of the row record"
```
Comment (ypc-faros) [inline sources/sheets-source/resources/schemas/sheets.json:13]: no description needed
This is the stream schema, it just defines the shape of the data returned by the stream

**Pull request merged**